server/task-workers/task-worker-client.cpp

```diff
@@ -0,0 +1,107 @@
+// Compiler for PHP (aka KPHP)
+// Copyright (c) 2020 LLC «V Kontakte»
+// Distributed under the GPL v3 License, see LICENSE.notice.txt
+
+#include <sys/param.h>
+#include <cassert>
+#include <cstdio>
+#include <unistd.h>
+
+#include "net/net-reactor.h"
+
+#include "server/php-queries.h"
+#include "server/task-workers/task-worker-client.h"
+#include "server/task-workers/task-workers-context.h"
+#include "server/task-workers/shared-context.h"
+
+int TaskWorkerClient::read_task_results(int fd, void *data __attribute__((unused)), event_t *ev) {
+  static int read_buf[PIPE_BUF / sizeof(int)];
```

Comment:
I suggest to create POD structure with named fields, and use it for writing and reading.